lms/djangoapps/oauth_dispatch/views.py

```diff
@@ -36,7 +36,11 @@ def get_adapter(self, request):
         """
         Returns the appropriate adapter based on the OAuth client linked to the request.
         """
-        if dot_models.Application.objects.filter(client_id=self._get_client_id(request)).exists():
+        dot_application = dot_models.Application.objects.filter(client_id=self._get_client_id(request))
+        if dot_application.exists():
+            if isinstance(self, RevokeTokenView):
+                request.POST = request.POST.copy()  # Make a mutable copy of the POST parameters.
+                request.POST['client_secret'] = dot_application[0].client_secret
```

Comment:
@nasthagiri Revoke token works only with client secret provided alongside client ID and we do not want mobile apps to store client secret and send in request, right?
So, I have implicitly added client secret to the request we are dispatching.